src/xmipp/libraries/data/pdb.cpp

```diff
@@ -483,15 +494,18 @@ void PDBRichPhantom::read(const FileName &fnPDB, double pseudoatoms, double thre
         REPORT_ERROR(ERR_IO_NOTEXIST, fnPDB);
 
     // Process all lines of the file
-    std::string line;
+    static const std::string empty(80, ' ');
+    auto line = std::string(80, ' ');
     std::string kind;
 
     RichAtom atom;
+    int i=0;
     while (!fh_in.eof())
     {
         // Read an ATOM line
         getline(fh_in, line);
-        if (line == "")
+        line.resize (80,' ');
+        if (line == empty)
```

Comment:
careful, before you were testing if the was nothing in the line, now you're testing if the line is full of spaces. Not the same

Reply:
Why not? If you add spaces to nothing and then compare it to the same number of space, then it should be, no?

Reply:
is it better to compare and then resize?

Reply:
I decided it is.  It would mean resizing less often and also remove the need for a variable for the empty line of spaces